### PR TITLE
Initialize feedstock

### PR DESCRIPTION
**Pull request by SC426**
Jira ticket: https://anaconda.atlassian.net/browse/PKG-701
Upstream repo: https://github.com/pdm-project/pdm

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -57,8 +59,14 @@ test:
 about:
   home: https://pypi.org/project/pdm/
```
Comment (cbouss) [inline recipe/meta.yaml]: I would use https://pdm.fming.dev/ for both home and doc_url and https://github.com/pdm-project/pdm for dev_url.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,6 +22,8 @@ requirements:
     - pip
     - python
     - pdm-pep517 >=0.3.0
```
Comment (cbouss) [inline recipe/meta.yaml]: Should be >=1.0
https://github.com/pdm-project/pdm/blob/2.2.1/pyproject.toml#L167

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,6 +22,8 @@ requirements:
...
+    - setuptools
+    - wheel
   run:
     - cachecontrol >=0.12.11
     - certifi >=2022.6.15
```
Comment (cbouss) [inline recipe/meta.yaml:31]: I don't see pip listed: https://github.com/pdm-project/pdm/blob/2.2.1/pyproject.toml#L10
importlib-metadata is for py<310: https://github.com/pdm-project/pdm/blob/2.2.1/pyproject.toml#L29

File: recipe/meta.yaml [hunk 4]
```diff
@@ -57,8 +59,14 @@ test:
 about:
```
Comment (cbouss) [inline recipe/meta.yaml:61]: You could add a test command using pdm like `- pdm --python`

You also miss an entrypoint in the top build section (see: https://github.com/pdm-project/pdm/blob/2.2.1/pyproject.toml#L54)

File: recipe/meta.yaml [hunk 5]
```diff
@@ -22,6 +22,8 @@ requirements:
     - pip
```
Comment (cbouss) [inline recipe/meta.yaml:24]: FYI version 2.3.1 is out: https://github.com/pdm-project/pdm/releases/tag/2.3.1

**Review by cbouss — CHANGES_REQUESTED** [review]

**Review by varlackc — COMMENTED** [review]
after the requested changes are made we might be able to give an approval.

File: recipe/meta.yaml (hunk 5 above)
Comment (SC426) [reply recipe/meta.yaml:24]: Are you ok leaving this at 2.2.1 for now? Updating to 2.3.1 will require building more dependencies.

File: recipe/meta.yaml (hunk 4 above)
Comment (SC426) [reply recipe/meta.yaml:61]: `- pdm --python` didn't seem to work, but I did add `- pdm --version`

Comment (cbouss) [reply recipe/meta.yaml:61]: That works too

File: recipe/meta.yaml (hunk 5 above)
Comment (cbouss) [reply recipe/meta.yaml:24]: ok, in that case 2.2.1 is fine.

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: I meant in the host section.
The host section should have:
```
host:
  - pdm-pep517 >=1.0
```
(https://github.com/pdm-project/pdm/blob/2.2.1/pyproject.toml#L167)
 And the run section should still have:
```
 run:
  - pdm-pep517 <2.0.0,>=1.0.0
```
(https://github.com/pdm-project/pdm/blob/2.2.1/pyproject.toml#L25)

**Review by cbouss — COMMENTED** [review]
Just the small fix to pdm-pep517 and I am good with it.

File: recipe/meta.yaml (hunk 2 above)
Comment (SC426) [reply recipe/meta.yaml]: Fixed this. Thanks!

**Review by cbouss — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**